src/nvim/channel.c

```diff
@@ -706,6 +706,7 @@ void channel_terminal_open(Channel *chan)
   topts.resize_cb = term_resize;
   topts.close_cb = term_close;
   Terminal *term = terminal_open(topts);
+  curbuf->b_p_channel = (long)chan->id;
```

Comment:
Should this option be set in `terminal_open()`, before event `TermOpen` is applied? 
Does the last section in the help for `terminal-status` needs to be updated after the channel id became an option?

```
These variables are initialized before TermOpen, so you can use them in
a local 'statusline'. Example: >
    :autocmd TermOpen * setlocal statusline=%{b:term_title}
```

Reply:
sounds like a good idea.